gpytorch/kernels/rbf_kernel.py

```diff
@@ -94,7 +95,18 @@ def __init__(
         )
 
     def forward(self, x1, x2, diag=False, **params):
-        x1_ = x1.div(self.lengthscale)
-        x2_ = x2.div(self.lengthscale)
-        diff = self._covar_dist(x1_, x2_, square_dist=True, diag=diag, dist_postprocess_func=postprocess_rbf, **params)
-        return diff
+        if (
+            x1.requires_grad
+            or x2.requires_grad
+            or (self.ard_num_dims is not None and self.ard_num_dims > 1)
+            or diag
+        ):
+            x1_ = x1.div(self.lengthscale)
+            x2_ = x2.div(self.lengthscale)
+            diff = self._covar_dist(x1_, x2_, square_dist=True, diag=diag, **params)
+            return postprocess_rbf(diff)
```

Comment:
@KeAWang Why does postprocess get called out of the JIT function now? The purpose of having this method at all was  was so that the full computation would be done inside a JIT script.

If we're not going to have postprocess scripts and only do the distance code in a JIT script, then the `dist_postprocess_func` arg needs to be removed everywhere and the function inlined here.